internal/addon/authentication/var.go

```diff
@@ -1,6 +1,8 @@
 package authentication
 
 const (
+	// ExistingSecret represents a pre-existing secret in the cluster.
+	ExistingSecret AuthenticationType = "ExistingSecret"
```

Comment:
note: We need to rename this after some doing some research to something else that indicates that it is user provided.

Reply:
Changed it to `SecretReference` since it's the wording used in similar cases in openshift/api